bot/exts/events/hacktoberfest/_utils.py

```diff
@@ -0,0 +1,452 @@
+    """
+    Extract shortname from https://api.github.com/repos/* URL.
+
+    e.g. "https://api.github.com/repos/python-discord/sir-lancebot"
+         |
+         V
+         "python-discord/sir-lancebot"
+    """
+    exp = r"https?:\/\/api.github.com\/repos\/([/\-\_\.\w]+)"
+    return re.findall(exp, in_url)[0]
+
+
+async def categorize_prs(bot: Bot, prs: list[dict]) -> tuple[list[dict], list[dict]]:
+    """
+    Categorize PRs into 'in_review' and 'accepted' and returns as a tuple.
+
+    PRs created less than 14 days ago are 'in_review', PRs that are not
+    are 'accepted' (after 14 days review period).
+
+    PRs that are accepted must either be merged, approved, or labelled
+    'hacktoberfest-accepted'.
+    """
+    now = datetime.now()
+    in_review = []
+    accepted = []
+    for pr in prs:
+        if (pr["created_at"] + timedelta(REVIEW_DAYS)) > now:
+            in_review.append(pr)
+        elif await is_accepted(bot, pr):
+            accepted.append(pr)
+
+    return in_review, accepted
+
+
+def build_prs_string(prs: list[dict], user: str) -> str:
+    """
+    Builds a discord embed compatible string for a list of PRs.
+
+    Repository name with the link to pull requests authored by 'user' for
+    each PR.
+    """
+    base_url = "https://www.github.com/"
+    str_list = []
+    repo_list = [pr["repo_shortname"] for pr in prs]
+    prs_list = Counter(repo_list).most_common(5)  # get first 5 counted PRs
+    more = len(prs) - sum(i[1] for i in prs_list)
+
+    for pr in prs_list:
+        # For example: https://www.github.com/python-discord/bot/pulls/octocat
+        # will display pull requests authored by octocat.
+        # pr[1] is the number of PRs to the repo
+        string = f"{pr[1]} to [{pr[0]}]({base_url}{pr[0]}/pulls/{user})"
+        str_list.append(string)
+    if more:
+        str_list.append(f"...and {more} more")
+
+    return "\n".join(str_list)
+
+
+def contributionator(n: int) -> str:
+    """Return "contribution" or "contributions" based on the value of n."""
+    if n == 1:
+        return "contribution"
+    else:
+        return "contributions"
+
+
+def author_mention_from_context(ctx: commands.Context) -> tuple[str, str]:
+    """Return stringified Message author ID and mentionable string from commands.Context."""
+    author_id = str(ctx.author.id)
+    author_mention = ctx.author.mention
+
+    return author_id, author_mention
+
+
+# Util functions for `.hacktoberfest issue`
+async def get_issues(ctx: commands.Context, option: str) -> Optional[dict]:
+    """Get a list of the python issues with the label 'hacktoberfest' from the Github api."""
+    if option == "beginner":
+        if (ctx.message.created_at.replace(tzinfo=None) - ctx.cog.cache_timer_beginner).seconds <= 60:
+            log.debug("using cache")
+            return ctx.cog.cache_beginner
+    elif (ctx.message.created_at.replace(tzinfo=None) - ctx.cog.cache_timer_normal).seconds <= 60:
+        log.debug("using cache")
```

Comment:
Not really helpful, could you add a bit more information on this logging statement. And we can bump such statements to `info` IMO.

Secondly, I don't think `debug` level logging statements are visible in the pydis production environment (Confirmed by joe).